topics/07_web_ui_testing/main_web_automation_with_playwright.md

```diff
@@ -1,3 +1,488 @@
+It is a part of many browsers particularly in Chrome, Firefox, Safari, and Edge
+
+It incorporates several panels
+For Web Automation the most important is Elements Panel:
+It contains DOM
+It allows viewing and edit the HTML structure and CSS styles of a webpage in real-time. You can change attributes, add styles, and see how these affect the page immediately.
+Layout and Box Model: Visualize elements’ layout properties like padding, margin, border, and dimensions.
+
+### Locators
+Locators are used to identify and interact with WebElements within a web page’s Document Object Model (DOM). They serve as the foundation for automating web application tests by allowing testers to perform actions such as clicking buttons, entering text, or verifying content.
+In Python locators are objects that are created by using /consuming string that define the place of element within DOM
+
+The syntax of creating locator depends on of used library and the Test Framework. Below are examples for Selenium and Python Playwright
+
+#### The most typical types of Locators
+##### ID Locator: 
+Targets elements with a unique id attribute. It's the fastest and most reliable locator if available.
+
+Examples for DOM:
+
+```xml
+<div>
+    <h1>Header 1</h1>
+    <ul> Menu 1
+        <li>Item 1</li>
+        <li class='item2'>Item 2</li>
+        <li id='item3'>Item 3</li>
+        <li name='item4'>Item 4</li>
+        <li input>Item 4</li>
+        <li>Item5</li>
+    </ul>
+    <input type="hidden" name="type" class="js-site-search-type-field">
+    <div class="example-class">
+        <a>Click Here</a>
+</div>
+```
+
+In Selenium:
+```python
+from selenium import webdriver
+
+driver = webdriver.Chrome()
+driver.get("https://example.com")
+element = driver.find_element("id", "item3")
+element.click()
+driver.quit()
+```
+
+In Playwright:
+```python
+from playwright.sync_api import sync_playwright
+
+with sync_playwright() as p:
+    browser = p.chromium.launch()
+    page = browser.new_page()
+    page.goto("https://example.com")
+    element = page.locator("#item3")
+    element.click()
+    browser.close()
+```
+
+##### Class Name Locator: 
+Matches elements by their class attribute, often used for elements styled similarly.
+Examples
+In Selenium:
+```python
+element = driver.find_element("class name", "item2")
+element.click()
+```
+In Playwright:
+```python
+element = page.locator(".item2")
+element.click()
+```
+##### Name Locator: 
+Uses the name attribute of elements, commonly used for form inputs.
+
+Examples
+In Selenium:
+```python
+element = driver.find_element("name", "item4")
+```
+In Playwright:
+```python
+element = page.locator('[name="item4"])'
+```
+
+##### Tag Name Locator:
+Locates elements based on their HTML tag (e.g., input, div).
+
+Examples
```

Comment:
This `Examples` word can be removed. Add a blank line instead. Do the same for the others.